### PR TITLE
Create task

### PR DESCRIPTION
**Pull request by kdelchev**
Create rake task to download and update D3 files automatically.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +21,10 @@ Please note that D3 is provided via the asset pipeline and you do *not* need to
 Then add it to your manifest file, most probably at `app/assets/javascripts/application.js`:
 
     //= require d3
+
+## Development
+
+To update D3 version in gem run rake task:
```
Comment (iblue) [inline README.md]: Should read "To update the D3 version contained in the gem, you can run the following rake task"

File: Rakefile [hunk 2]
```diff
@@ -1,2 +1,13 @@
...
 Bundler::GemHelper.install_tasks
+
+namespace :d3 do
+  desc 'Update d3 version'
+  task :update_version do
+    `curl -o app/assets/javascripts/d3.js https://raw.githubusercontent.com/mbostock/d3/master/d3.js`
+    `curl -o app/assets/javascripts/d3.min.js https://raw.githubusercontent.com/mbostock/d3/master/d3.min.js`
+    `cp app/assets/javascripts/d3.js app/assets/javascripts/d3.v3.js`
+    `cp app/assets/javascripts/d3.min.js app/assets/javascripts/d3.v3.min.js`
+    puts "Update manually version to #{`grep 'version: ".*"' app/assets/javascripts/d3.js | cut -d '"' -f 2`}CHANGELOG.md\nREADME.md\nlib/d3/rails/version.rb\n"
```
Comment (iblue) [inline Rakefile]: Should read "Please update the version to ... manually in the following files"

**Comment by kdelchev** [comment]
Done!
:beer:

**Comment by iblue** [comment]
Thank you!

**Comment by ian-s-mcb** [comment]
Did this feature ever work? I submitted #35 because the `update_version` Rake task doesn't work for me.

**Comment by kdelchev** [comment]
Yes, it was working the previous time I was using it to update the version. Not sure if anything changed.

**Comment by ian-s-mcb** [comment]
@kdelchev Is it odd that the Rakefile within the d3-rails-3.5.6 Gem folder doesn't match Rakefile in the Gem's GitHub repo?

**Comment by kdelchev** [comment]
@ian-s-mcb it seems the gem was not correctly bundled. @iblue could you check it and confirm?

I would suggest to use directly the master branch
```ruby
gem 'd3-rails', github: 'iblue/d3-rails'
```
until the problem is solved. Or better fork and use your own fork to ensure you are using the current state.

**Comment by ian-s-mcb** [comment]
@kdelchev Same situation as before. I ran this:

```bash
$ gem uninstall d3-rails
# add the line you suggested to my GemFile
$ bundle install
$ bundle exec rake d3:update_version
```

and got:

```bash
rake aborted!
Don't know how to build task 'd3:update_version'

(See full trace by running task with --trace)
```

I did notice that d3-rails gem ended up being placed into a different directory:

```bash
$ bundle show d3-rails
/home/ian/.rvm/gems/ruby-2.2.0/bundler/gems/d3-rails-1450cbbf7452
```

And the Rakefile in that folder does match the one in the GitHub repo.

Sorry to bug you about this. I'm not experienced enough with these technologies to come up with solution on my own.

**Comment by kdelchev** [comment]
@ian-s-mcb I suppose you are adding the gem inside Rails project and want to execute the rake task inside the Rails project. Right?

The task is intended to be run inside the gem to update the javascript version of D3 that the gem is using. I run the task and produced https://github.com/iblue/d3-rails/pull/36 within 2 minutes.

Sorry this takes you so much time. It would be really helpful if you can add pull request updating the task description to be clear what it is actually doing.

**Comment by iblue** [comment]
Sorry I kept you waiting. Will look into it within 36 hours, I promise. Thanks for investigating so far!

**Comment by iblue** [comment]
So, as far as I understand @ian-s-mcb is trying to run `rake d3:update_version` in the application directory. This does not work, because `d3-rails` does not do anything with the applications Rakefile. Instead the rake task is for upgrading the d3 version in the gem (for developers of the gem). You have to run it in the gems directory and then submit a pull request. Maybe I should add this to the Readme...

**Comment by ian-s-mcb** [comment]
@iblue Understood. Thank you for clarifying the situation.

**Pull request merged**